cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1607,24 +1607,47 @@ folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobu
 }
 
 folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_metadata_async(
```

Comment:
I think this function should return
`folly::Future<std::pair<VariantKey, std::optional<google::protobuf::Any>>>`
as the `VariantKey` is never `std::nullopt` due to the check.
This will also simplify the logic in `PythonVersionStore::batch_read_metadata`

Reply:
good point